### PR TITLE
Technique example: data loader, Python to parquet

### PR DESCRIPTION
**Pull request by allisonhorst**
Deployed: https://observablehq.observablehq.cloud/framework-example-loader-python-to-parquet/

### CODE REVIEW

**Comment by Fil** [comment]
can we link to https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html ; and maybe  show a few useful options (like compression)?

**Review by Fil — APPROVED** [review]

File: examples/loader-python-to-parquet/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[Framework examples →](../)
+
+# Python data loader to generate Apache Parquet
+
+View live: <https://observablehq.observablehq.cloud/framework-example-loader-python-to-parquet/>
+
+This Observable Framework example demonstrates how to write a Python data loader that outputs an Apache Parquet file using the [pyarrow](https://pypi.org/project/pyarrow/) library. The loader reads in a CSV with records for over 91,000 dams in the United States from the [National Inventory of Dams](https://nid.sec.usace.army.mil/#/), selects several columns, then writes the data frame as a parquet file to standard output. The data loader lives in [`src/data/us-dams.parquet.py`](./src/data/us-dams.parquet.py).
```
Comment (Fil) [inline examples/loader-python-to-parquet/README.md:7]:
```suggestion
This Observable Framework example demonstrates how to write a Python data loader that outputs an Apache Parquet file using the [pyarrow](https://pypi.org/project/pyarrow/) library. The loader reads in a CSV with records for over 91,000 dams in the United States from the [National Inventory of Dams](https://nid.sec.usace.army.mil/), selects several columns, then writes the data frame as a parquet file to standard output. The data loader lives in [`src/data/us-dams.parquet.py`](./src/data/us-dams.parquet.py).
```

File: examples/loader-python-to-parquet/src/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Python data loader to generate Apache Parquet
+
+Here’s a Python data loader that accesses records for over 91,000 dams from the [National Inventory of Dams](https://nid.sec.usace.army.mil/#/), limits the data to only four columns, then outputs an Apache Parquet file to standard out.
```
Comment (Fil) [inline examples/loader-python-to-parquet/src/index.md:3]:
```suggestion
Here’s a Python data loader that accesses records for over 91,000 dams from the [National Inventory of Dams](https://nid.sec.usace.army.mil/), limits the data to only four columns, then outputs an Apache Parquet file to standard out.
```

File: examples/loader-python-to-parquet/src/index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const dams = FileAttachment("./data/us-dams.parquet").parquet();
+```
+
+We can display the table using `Inputs.table`.
+
+```js echo
+Inputs.table(dams)
+```
+
+Lastly, we can pass the table to `Plot.plot` to make a simple bar chart of dam counts by purpose, with color mapped to hazard classification.
```
Comment (Fil) [inline examples/loader-python-to-parquet/src/index.md:51]:
```suggestion
Lastly, we can pass the table to Observable Plot to make a simple bar chart of dam counts by purpose, with color mapped to hazard classification.
```

**Comment by allisonhorst** [comment]
@Fil I added the compression codec explicitly in the loader (compression="snappy"), and include a sentence pointing to the write_table docs and different compression algorithms. Look okay? 

> can we link to https://arrow.apache.org/docs/python/generated/pyarrow.parquet.write_table.html explicitly mentioning that there are many options (and recommend compression); and maybe show compression in action?

**Review by mbostock — COMMENTED** [review]
Mind copying the new virtual environment pattern from #1468?

````md
<div class="note">

To run this data loader, you’ll need python3 and the geopandas, matplotlib, io, and sys modules installed and available on your `$PATH`.

</div>

<div class="tip">

We recommend using a [Python virtual environment](https://observablehq.com/framework/loaders#venv), such as with venv or uv, and managing required packages via `requirements.txt` rather than installing them globally.

</div>
````

**Comment by jaanli** [comment]
Quick question - would dbt work here?

File: examples/loader-python-to-parquet/src/index.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+```js echo
+Plot.plot({
+  marginLeft: 220,
+  color: {legend: true, domain: ["Undetermined", "Low", "Significant", "High"]},
+  marks: [
+    Plot.barX(dams,
+      Plot.groupY({x: "count"}, {y: "Primary Purpose", fill: "Hazard Potential Classification", sort: {y: "x", reverse: true}
+      })
+    )
```
Comment (mbostock) [inline examples/loader-python-to-parquet/src/index.md:69]: Please prettier this. 🙏

Also, you can use `sort: {y: "-x"}` to shorten.

Comment (allisonhorst) [reply examples/loader-python-to-parquet/src/index.md:69]: Yep will do!

File: examples/loader-python-to-parquet/src/index.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ const dams = FileAttachment("data/us-dams.parquet").parquet();
 We can display the table using `Inputs.table`.
 
 ```js echo
-Inputs.table(dams)
+Inputs.table(dams);
```
Comment (mbostock) [inline examples/loader-python-to-parquet/src/index.md]: This prettier edit will prevent the table from displaying.

Comment (allisonhorst) [reply examples/loader-python-to-parquet/src/index.md]: I think this was correctly updated in a subsequent commit (I turned prettier off after formatting, to leave the semicolon after FileAttachment but remove in the Inputs.table and Plot code: 

```js echo
Inputs.table(dams)
```

**Pull request merged**